bin/migrate-oats-data/applications/decisions/application_decisions_init.py

```diff
@@ -0,0 +1,202 @@
+                decision_description = COALESCE(EXCLUDED.decision_description, alcs.application_decision.decision_description),
+                rescinded_comment =  COALESCE(EXCLUDED.rescinded_comment, alcs.application_decision.rescinded_comment),
+                rescinded_date =  COALESCE(EXCLUDED.rescinded_date, alcs.application_decision.rescinded_date),
+                is_subject_to_conditions = COALESCE(EXCLUDED.is_subject_to_conditions, alcs.application_decision.is_subject_to_conditions),
+                oats_alr_appl_decision_id = EXCLUDED.oats_alr_appl_decision_id;
+    """
+    return query
+
+
+def _prepare_oats_alr_applications_data(row_data_list):
+    data_list = []
+    for row in row_data_list:
+        resolution_number, year = _map_resolution_number_and_year(row)
+        mapped_row = {
+            "audit_created_by": OATS_ETL_USER,
+            "created_at": add_timezone_and_keep_date_part(row.get("when_created")),
+            "date": add_timezone_and_keep_date_part(row.get("decision_date")),
+            "decision_description": row.get("decision_desc"),
+            "decision_maker": "CEO",
+            "is_draft": False,
+            "outcome_code": _map_outcome_code(row),
+            "rescinded_comment": row.get("rescinded_comments"),
+            "rescinded_date": add_timezone_and_keep_date_part(
+                row.get("rescinded_date")
+            ),
+            "resolution_number": resolution_number,
+            "resolution_year": year,
+            "was_released": True,
+            "application_uuid": row.get("app_uuid"),
+            "is_subject_to_conditions": _map_is_subject_to_conditions(row),
+            "oats_alr_appl_decision_id": row.get("alr_appl_decision_id"),
+        }
+        data_list.append(mapped_row)
+
+    return data_list
+
+
+def _map_outcome_code(row):
+    if row.get("rescinded_date", None):
+        return "RESC"
+    elif row.get("outright_refusal_ind", None) == "N":
+        return "APPR"
+    elif row.get("outright_refusal_ind", None) == "Y":
+        return "REFU"
+    return None
+
+
+def _map_resolution_number_and_year(row):
+    year_and_number = str(row["resolution_number"])
+    # some data appears to only have a year, if this is the case the number will be set to 0
+    year_and_number_length = len(year_and_number)
+    if year_and_number_length == 4:
+        resolution_number = 0
+    else:
+        resolution_number = int(year_and_number[:-4])
+    year = int(year_and_number[-4:])
+    return resolution_number, year
+
+
+def _map_is_subject_to_conditions(row):
+    return row.get("is_subject_to_conditions", "False") == True
+
+
+@inject_conn_pool
+def clean_application_decisions(conn=None):
+    logger.info("Start application_decisions cleaning")
+    with conn.cursor() as cursor:
+        cursor.execute(
+            f"DELETE FROM alcs.application_decision WHERE audit_created_by = '{OATS_ETL_USER}' AND audit_updated_by is NULL"
```

Comment:
So I've been wondering about this. What is the purpose of checking the audit_updated_by is NULL if we are not going to actually clean anything on production, and this is mostly for our local development? Is it used to avoid data that was manually changed in local db?

Reply:
That was the original idea, to avoid deleting manually changed data as it can cause FK constraints in certain circumstances. I am not sure it is critical, we could probably look into removing it.

Reply:
Nope, all good, just wanted to clarify if it is related somehow to prod data or a result of local development. I think it is a good idea to have it to avoid random errors on local